src/api/java/AST.java

```diff
@@ -28,23 +28,16 @@
      * Object comparison.
      * 
      * @param o another AST
-     **/    
+     **/
+    @Override
     public boolean equals(Object o)
     {
-        AST casted = null;
+        if (o == this) return true;
+        if (!(o instanceof AST)) return false;
+        AST casted = (AST) o;
 
-        try
-        {
-            casted = AST.class.cast(o);
-        } catch (ClassCastException e)
-        {
-            return false;
-        }
-
-        return  (this == casted) ||
-                (this != null) &&
-                (casted != null) &&
-                (getContext().nCtx() == casted.getContext().nCtx()) &&
```

Comment:
`o` can be an object other than an AST (at least in older versions of Java, but perhaps not in 1.8). I remember adding those casts for that particular reason. What is the difference between `o instanceof AST` and `try {AST.class.cast(o)} catch ...`? Will this behave the same on all platforms and through all versions of Java? If not, then I'd rather keep the old code instead of spending a lot of time on testing the new code.